### PR TITLE
Update ruff to v6.8 and many typos

### LINKED ISSUE #2547: update ruff to v6.8
see MR #2548

### PR DESCRIPTION
**Pull request by gdevos010**
Checklist before merging this PR:
- [x] Mentioned all issues that this PR fixes or addresses.
- [x] Summarized the updates of this PR under **Summary**.
- [x] Added an entry under **Unreleased** in the [Changelog](../CHANGELOG.md).

Fixes #2547.

### Summary
update to ruff v6.8 
fixed many typos

### CODE REVIEW

File: examples/02-data-processing.ipynb
Comment (madtoinou) [inline examples/02-data-processing.ipynb:1]: Please revert the changes to the images

File: examples/03-FFT-examples.ipynb
Comment (madtoinou) [inline examples/03-FFT-examples.ipynb:1]: Please revert the changes to the images

File: examples/11-GP-filter-examples.ipynb
Comment (madtoinou) [inline examples/11-GP-filter-examples.ipynb:1]: Revert changes to images

File: pyproject.toml
Comment (madtoinou) [inline pyproject.toml:1]: Why did you remove that?

**Review by madtoinou — CHANGES_REQUESTED** [review]
Thanks @gdevos010 for this PR, looks good overall.

Can you please revert the changes to the images in the notebooks? Also, found some changes in the code that might cause some issues.

Also, we need to make sur follow the migrations instruction from 0.3.6 to 0.6.8 to avoid issues.

File: pyproject.toml
Comment (gdevos010) [reply pyproject.toml:1]: It's deprecated in v6.8

**Comment by madtoinou** [comment]
Replaced by #2589.

**Pull request closed without merging**